testutil/src/main/java/org/spine3/test/NullToleranceTest.java

```diff
@@ -0,0 +1,263 @@
+                                                                .equals(preconditionsElement.getClassName());
+        if (!isPreconditionsClass) {
+            return false;
+        }
+
+        final StackTraceElement expectedUtilClassElement = stackTraceElements[1];
+        final boolean isCorrectMethod = isCorrectMethodName(methodName, expectedUtilClassElement);
+        if (!isCorrectMethod) {
+            return false;
+        }
+
+        final boolean isUtilClass = utilClass.getName()
+                                             .equals(expectedUtilClassElement.getClassName());
+        return isUtilClass;
+    }
+
+    private boolean isCorrectMethodName(String methodName, StackTraceElement expectedUtilClassElement) {
+        final boolean isCorrectMethodName = methodName.equals(expectedUtilClassElement.getMethodName());
+        return isCorrectMethodName;
+    }
+
+    /**
+     * Creates a new builder for the {@code NullToleranceTest}.
+     *
+     * @return the {@code Builder}
+     */
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    private void initPrimitiveValuesSet() {
+        primitiveValues.add(boolean.class);
+        primitiveValues.add(byte.class);
+        primitiveValues.add(short.class);
+        primitiveValues.add(int.class);
+        primitiveValues.add(long.class);
+        primitiveValues.add(char.class);
+        primitiveValues.add(float.class);
+        primitiveValues.add(double.class);
+    }
+
+    /**
+     * A builder for producing the {@link NullToleranceTest} instance.
+     */
+    public static class Builder {
+
+        private Class utilClass;
+        private Set<String> excludedMethods;
+        private Map defaultValues;
+
+        private Builder() {
+            defaultValues = newHashMap();
+            excludedMethods = newHashSet();
+        }
+
+        /**
+         * Sets the utility class.
+         *
+         * @param utilClass the utility {@link Class}
+         * @return the {@code Builder}
+         */
+        public Builder setClass(Class utilClass) {
+            this.utilClass = utilClass;
+            return this;
+        }
+
+        /**
+         * Adds the method name in the {@code excludedMethods} set.
+         *
+         * @param methodName the name of the excluded method
+         * @return the {@code Builder}
+         */
+        public Builder excludeMethod(String methodName) {
+            excludedMethods.add(methodName);
+            return this;
+        }
+
+        /**
+         * Adds the default value for the class in the {@code defaultValues} map.
+         *
+         * @param clazz the class for which will be added default value
+         * @param value the default value for the class
+         * @return the {@code Builder}
+         */
+        @SuppressWarnings("unchecked") // check on the method level.
+        public <I> Builder addDefaultValue(Class<I> clazz, I value) {
```

Comment:
Why do we need the first parameter at all (assuming we can get a class from non-null passed value)?